### PR TITLE
Refactor osdctl cluster health

### PR DESCRIPTION
**Pull request by fahlmant**
Refactors `osdctl cluster health` to remove dependence on k8sclusterresourcefactory

### CODE REVIEW

File: cmd/cluster/health.go [hunk 1]
```diff
@@ -3,74 +3,59 @@ package cluster
...
+	clusterID  string
+	output     string
+	verbose    bool
+	awsProfile string
 }
 
 // newCmdHealth implements the health command to describe number of running instances in cluster and the expected number of nodes
 func newCmdHealth(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *cobra.Command {
-	ops := newHealthOptions(streams, flags, globalOpts)
+	ops := newHealthOptions()
```
Comment (dkeohane) [inline cmd/cluster/health.go:34]: Can remove params from newCmdHealth

Comment (fahlmant) [reply cmd/cluster/health.go:34]: Tbh I thought I did, doing it now

File: cmd/cluster/health.go [hunk 2]
```diff
@@ -3,74 +3,55 @@ package cluster
...
 
-func newHealthOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, globalOpts *globalflags.GlobalOptions) *healthOptions {
-	return &healthOptions{
-		k8sclusterresourcefactory: k8spkg.ClusterResourceFactoryOptions{
-			Flags: flags,
-		},
-		IOStreams:     streams,
-		GlobalOptions: globalOpts,
-	}
+func newHealthOptions() *healthOptions {
```
Comment (dkeohane) [inline cmd/cluster/health.go:49]: minor: Can probably remove this func now

**Review by dkeohane — APPROVED** [review]
/lgtm
Just 1 minor comment, but looks good

File: cmd/cluster/health.go (hunk 2 above)
Comment (fahlmant) [reply cmd/cluster/health.go:49]: I'm leaving it for now so it matches all our commands. I'm planning on cleaning these up across all commands as part of the larger refactor eventually.

**Pull request merged**